src/main/java/com/coveo/spillway/storage/RedisStorage.java

```diff
@@ -69,15 +72,21 @@
 
   private static final String KEY_SEPARATOR_SUBSTITUTE = "_";
   private static final String WILD_CARD_OPERATOR = "*";
-  private static final String COUNTER_SCRIPT =
-      "local counter = redis.call('INCRBY', KEYS[1], ARGV[1]); "
-          + "if counter  > tonumber(ARGV[2]) + tonumber(ARGV[1])"
-          + "then counter = redis.call('INCRBY', KEYS[1], -ARGV[1]) "
-          + "end "
-          + "return tostring(counter)";
-
+  private static final String COUNTER_SCRIPT = getCounterLuaScript();
   private final JedisPool jedisPool;
   private final String keyPrefix;
+  private static final String COUNTER_LUA_SCRIPT = "com/coveo/spillway/storage/redisCounter.lua";
+
+  private static String getCounterLuaScript() throws SpillwayLuaResourceNotFoundException {
+    try {
+      return IOUtils.toString(
+          RedisStorage.class.getClassLoader().getResourceAsStream(COUNTER_LUA_SCRIPT),
+          StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      throw new SpillwayLuaResourceNotFoundException(
+          "Error reading Resource: " + COUNTER_LUA_SCRIPT, e);
```

Comment:
A few things here:

1. exception name is probably wrong - it does not necessarily mean that the resource was not found (though it is likely), only that it was not able to read it
2. message should match exception name in some way IMO
3. you could probably take only the script path and the exception as params and have the rest of the message inside the constructor

Reply:
👍